include/zim/export_api.h

```diff
@@ -0,0 +1,10 @@
+#ifndef ZIM_EXPORT_API_H
+#define ZIM_EXPORT_API_H
+
+#if (defined _WIN32 || defined __CYGWIN__) && defined ENABLE_EXPORT_DLL
+    #define LIBZIM_API __declspec(dllexport)
+#else
+    #define LIBZIM_API
+#endif
+
+#endif
```

Comment:
I would move that in `zim.h` header (where there is also the `DEPRECATED` definition).

And it would avoid new `#include "export_api.h"` as most (if not all) header files already include `zim.h`.

Reply:
but not every file include zim.h

Reply:
In header that doesn't include zim.h directly nor indirectly, yes.
It should be easily detectable as the compilation will fail if `zim.h` is not included in the include chain.